doc/tutorials/klondike/app/lib/step5/components/stock_pile.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:ui';
+
+import 'package:flame/components.dart';
+import 'package:flame/events.dart';
+
+import '../klondike_game.dart';
+import '../pile.dart';
+import 'card.dart';
+import 'waste_pile.dart';
+
+class StockPile extends PositionComponent
+    with TapCallbacks, HasGameReference<KlondikeGame>
+    implements Pile {
+  StockPile({super.position}) : super(size: KlondikeGame.cardSize);
+
+  /// Which cards are currently placed onto this pile. The first card in the
+  /// list is at the bottom, the last card is on top.
+  final List<Card> _cards = [];
+
+  //#region Pile API
+
+  @override
+  // Dummy - not used.
+  Vector2 dropPosition() => position;
```

Comment:
You can throw an `UnimplementedException` here

Reply:
Done, in both StockPile and WastePile classes. Used `UnsupportedError`: those classes can never accept dropped cards. It's against the rules of the game for them.